nomad/server.go

```diff
@@ -1192,8 +1192,8 @@ func (s *Server) setupConsul(consulConfigFunc consul.ConfigAPIFunc, consulACLs c
 // setupVaultClient is used to set up the Vault API client.
 func (s *Server) setupVaultClient() error {
 	vconfig := s.config.GetDefaultVault()
-	if vconfig != nil && vconfig.DefaultIdentity != nil {
-		s.vault = &NoopVault{}
+	if vconfig != nil && vconfig.Addr == config.DefaultVaultAddr && vconfig.Token == "" {
```

Comment:
We unfortunately can't check for an empty address string here since it's always set to this default. But I would argue that it's kind of a bad default, and I expect people that are actually using Vault to change this default address.

We must check both values because some environments may be deploy Vault "tokenless" (https://github.com/hashicorp/nomad/issues/19380).

Reply:
Just to make sure that playing around in dev mode works with the legacy workflow and an env var for `VAULT_TOKEN`, we probably should leave the `DefaultIdentity` condition in there and just add the new ones. So:

```suggestion
	if vconfig != nil && vconfig.DefaultIdentity != nil && vconfig.Addr == config.DefaultVaultAddr && vconfig.Token == "" {
```

Reply:
Yeah that makes sense. There's no such thing as Vault without ACLs.